src/main/java/teammates/ui/automated/FeedbackSessionPublishedEmailParticularStudentsWorkerAction.java

```diff
@@ -0,0 +1,63 @@
+package teammates.ui.automated;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import teammates.common.datatransfer.attributes.FeedbackSessionAttributes;
+import teammates.common.datatransfer.attributes.InstructorAttributes;
+import teammates.common.datatransfer.attributes.StudentAttributes;
+import teammates.common.exception.TeammatesException;
+import teammates.common.util.Const.ParamsNames;
+import teammates.common.util.EmailWrapper;
+import teammates.common.util.Logger;
+import teammates.logic.api.EmailGenerator;
+
+/**
+ * Task queue worker action: sends feedback session reminder email to particular students of a course.
+ */
+public class FeedbackSessionPublishedEmailParticularStudentsWorkerAction extends AutomatedAction {
+
+    private static final Logger log = Logger.getLogger();
+
+    @Override
+    protected String getActionDescription() {
+        return null;
+    }
+
+    @Override
+    protected String getActionMessage() {
+        return null;
+    }
+
+    @Override
+    public void execute() {
+        String feedbackSessionName = getNonNullRequestParamValue(ParamsNames.SUBMISSION_FEEDBACK);
+        String courseId = getNonNullRequestParamValue(ParamsNames.SUBMISSION_COURSE);
+        String[] usersToRemind = getNonNullRequestParamValues(ParamsNames.SEND_PUBLISHED_EMAIL);
+
+        try {
+            FeedbackSessionAttributes session = logic.getFeedbackSession(feedbackSessionName, courseId);
+            List<StudentAttributes> studentsToEmailList = new ArrayList<>();
+            List<InstructorAttributes> instructorsToEmailList = new ArrayList<>();
+
+            for (String userEmail : usersToRemind) {
+                StudentAttributes student = logic.getStudentForEmail(courseId, userEmail);
+                if (student != null) {
+                    studentsToEmailList.add(student);
+                }
+
+                InstructorAttributes instructor = logic.getInstructorForEmail(courseId, userEmail);
```

Comment:
Would it be more efficient if we filter through the instructorDB only when the userEmail cannot be found in the studentDB?

Reply:
This I think is because an instructor may also be registered as a student at the same time (#5770) therefore we need to iterate both lists.